algorithms/threading/threading.cpp

```diff
@@ -70,7 +73,7 @@ DAAL_EXPORT size_t _setNumberOfThreads(const size_t numThreads, void ** init)
     if (numThreads != 0)
     {
         _daal_tbb_task_scheduler_free(*init);
-        *init = (void *)(new tbb::task_scheduler_init(numThreads));
+        *init = static_cast<void *>(new tbb::global_control(tbb::global_control::max_allowed_parallelism, numThreads));
```

Comment:
init is wrong name for the parameter after your changes. It also applicable for corresponding field name